### ISSUE TITLE
Documentation inconsistency regarding randomized startup delay

### ISSUE THREAD

**Issue by joekohlsdorf** [rabbitmq/rabbitmq-peer-discovery-k8s#22]
According to the documentation this plugin uses a randomized startup delay:
> This backend relies on randomized startup delay to reduce the probability of a race condition during initial cluster formation (see below).

The logs of node startup don't support this statement:
```
2018-04-10 16:53:01.887 [info] &lt;0.189.0&gt; Peer discovery backend does not support locking, falling back to randomized delay
2018-04-10 16:53:01.888 [info] &lt;0.189.0&gt; Peer discovery backend rabbit_peer_discovery_k8s does not support registration, skipping randomized startup delay.
```

And if the only supported method of running a RabbitMQ cluster in Kubernetes is a StatefulSet, a randomized startup delay isn't necessary because Kubernetes will start Pods one after another (taken from the documentation):
> The StatefulSet controller starts Pods one at a time, in order by their ordinal index. It waits until each Pod reports being Ready before starting the next one.

**Comment by michaelklishin** [comment]
Would you mind linking me to the exact [paragraph in the docs](https://github.com/rabbitmq/rabbitmq-website/blob/live/site/cluster-formation.xml) that you find inconsistent?

`StatefulSet` is the only solution I see as practical but we don't enforce it.
Randomized startup delay can be set to, say, a value between 0 and 1 second in case it's not necessary. We can mention that in the docs.

**Comment by michaelklishin** [comment]
Also a link to the Kubernetes docs guide you're quoting would be nice, we then would link to it from ours.

**Comment by joekohlsdorf** [comment]
Here you go:
* [RabbitMQ documentation](https://github.com/rabbitmq/rabbitmq-website/blob/724cabbde8cbfbd006f87506d905b47a23891c65/site/cluster-formation.xml#L421)
* [Kubernetes StatefulSet documentation](https://kubernetes.io/docs/tasks/run-application/run-replicated-stateful-application/#understanding-stateful-pod-initialization)

> Randomized startup delay can be set to, say, a value between 0 and 1 second in case it's not necessary. We can mention that in the docs.

According to the log which I showed startup delay currently isn't working, that is the inconsistency between the documentation and observed behaviour which I want to report. It is either a bug in the plugin or a problem with the documentation, depending if a startup delay is wanted or not.

With a StatefulSet a startup delay isn't necessary, I therefore included that information. I'm not sure how you could run a cluster with this plugin in any other way because of the dependency on a headless service. So in my opinion the delay could be disabled by default and enabled with a flag if needed for special cases.

**Comment by michaelklishin** [comment]
Closing in favor of #23 and assorted doc updates that will reference this issue. Thanks.

**Comment by michaelklishin** [comment]
What I will do in #23 is making randomized startup delay work (currently by indicating that this plugin supports registration, which is not really true) but reducing the range if we can do it without ugly hacks (the server may call plugin callbacks later than we need in this case).

There's also a doc example now that demonstrates how to do it manually.

That's been a very useful observation and set of recommendations, thank you @joekohlsdorf.
